messaging/src/main/java/org/axonframework/messaging/MessageStream.java

```diff
@@ -25,84 +27,183 @@
 import java.util.function.Function;
 import java.util.function.Predicate;
 import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
 
 /**
- * Represents a stream of {@link Message Messages} that can be consumed as they become available.
+ * Represents a stream of {@link Entry entries} containing {@link Message Messages} of type {@code M} that can be
+ * consumed as they become available.
  *
- * @param <M> The type of {@link Message} carried in this stream.
+ * @param <M> The type of {@link Message} contained in the {@link Entry entries} of this stream.
  * @author Allard Buijze
+ * @author Ivan Dugalic
+ * @author Milan Savić
+ * @author Mitchell Herrijgers
  * @author Steven van Beelen
  * @since 5.0.0
  */
 public interface MessageStream<M extends Message<?>> {
```

Comment:
:thinking: doesn't that make sense if there is no `MessageStream#asPublisher` operation yet? Or, do you mean there are no tests validating that wrapping a `Publisher` implementation like `Flux` works as expected in the `MessageStream`?

Reply:
I was referring to https://github.com/reactive-streams/reactive-streams-jvm/tree/master/tck. And since we want our `MessageStream` to extend `Publisher`, I think that using the TCK is a must.

Reply:
Ah yes, gotcha. As with the other reactive points, I've added this pointer to #3155. Thanks for clarifying!